test/e2e/prometheus_test.go

```diff
@@ -1568,79 +1621,321 @@ func testOperatorNSScope(t *testing.T) {
 
-type prometheusQueryAPIResponse struct {
-	Status string       `json:"status"`
-	Data   *queryResult `json:"data"`
-}
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+	ctx.SetupPrometheusRBAC(t, ns, framework.KubeClient)
+	name := "test"
 
-func basicQueryWorking(ns, svcName string) (bool, error) {
-	response, err := framework.QueryPrometheusSVC(ns, svcName, "/api/v1/query", map[string]string{"query": "up"})
+	//
+	// Setup sample app.
+	//
+
+	cert, err := ioutil.ReadFile("../../test/instrumented-sample-app/certs/cert.pem")
 	if err != nil {
-		return false, err
+		t.Fatalf("failed to load cert.pem: %v", err)
 	}
 
-	rq := prometheusQueryAPIResponse{}
-	if err := json.NewDecoder(bytes.NewBuffer(response)).Decode(&rq); err != nil {
-		return false, err
+	key, err := ioutil.ReadFile("../../test/instrumented-sample-app/certs/key.pem")
+	if err != nil {
+		t.Fatalf("failed to load key.pem: %v", err)
 	}
 
-	if rq.Status != "success" && rq.Data.Result[0].Value[1] == "1" {
-		log.Printf("Query Response not successful.")
-		return false, nil
+	tlsCertsSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Data: map[string][]byte{
+			"cert.pem": cert,
+			"key.pem":  key,
+		},
+	}
+
+	if _, err := framework.KubeClient.CoreV1().Secrets(ns).Create(tlsCertsSecret); err != nil {
+		t.Fatal(err)
+	}
+
+	simple, err := testFramework.MakeDeployment("../../test/framework/ressources/basic-auth-app-deployment.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	simple.Spec.Template.Spec.Containers[0].Args = []string{"--cert-path=/etc/certs"}
+
+	simple.Spec.Template.Spec.Volumes = []v1.Volume{
+		v1.Volume{
+			Name: "tls-certs",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: tlsCertsSecret.Name,
+				},
+			},
+		},
+	}
+
+	simple.Spec.Template.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
+		{
+			Name:      simple.Spec.Template.Spec.Volumes[0].Name,
+			MountPath: "/etc/certs",
+		},
+	}
+
+	if err := testFramework.CreateDeployment(framework.KubeClient, ns, simple); err != nil {
+		t.Fatal("Creating simple basic auth app failed: ", err)
+	}
+
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"group": name,
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeLoadBalancer,
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Name: "web",
+					Port: 8080,
+				},
+				v1.ServicePort{
+					Name: "mtls",
+					Port: 8081,
+				},
+			},
+			Selector: map[string]string{
+				"group": name,
+			},
+		},
+	}
+
+	if _, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, svc); err != nil {
+		t.Fatal(err)
+	}
+
+	//
+	// Setup monitoring.
+	//
+
+	sm := framework.MakeBasicServiceMonitor(name)
+	sm.Spec.Endpoints = []monitoringv1.Endpoint{
+		{
+			Port:     "mtls",
+			Interval: "30s",
+			Scheme:   "https",
+			TLSConfig: &monitoringv1.TLSConfig{
+				InsecureSkipVerify: true,
```

Comment:
Is this still validating authorities?

Reply:
The test does assure, that Prometheus sends properly signed requests to the sample app. It does *not* ensure, that the responses are properly signed by the sample app.

While the latter check would be a great addition, the test is already valuable, given the first check.

I guess we would need to generate certificates on the fly to ensure, that the domain name in the certificates is correct. Any idea how we could get around this?